Added expanded property to show inline div on row click event which can be used as a container to display any other component

ramanav opened the pull request:
# Pull Request

Hi,

Thanks for helping us improve the Predix UI platform by submitting a Pull Request.

To help us merge your Pull Request as fast as possible, please fill out the following sections:

* ## A description of the changes proposed in the pull request:
We have a requirement to show inline px-data-table on row click of parent px-data-table. So i have added a div container to that can container web component. It will be hidden or displayed on row click event. And this row click event is enabled/disabled by an expanded property.

* ## A reference to a related issue (if applicable):
This is a generic requirement as well as we need this in our project.


* ## @mentions of the person or team responsible for reviewing proposed changes:
Please review my changes so that it can be merged into master.

* ## working tests:
#### The tests need to be functional and/or unit tests, written for the wct framework, and placed in the test folder, following our testing guidelines.
…an be used as a container to display any other component

Added expanded property to show inline div on row click event which can
be used as a container to display any other component

mdwragg commented:
Hi, sadly this has caused some unit tests to fail, can you please verify that the tests are running locally?

Speaking of, can you please add unit tests to cover your change? Thanks!

ramanav commented:
Really sorry for breking test cases. I have fixed those test cased and added one more for row click.
Can you please review and let me know if anything else is needed.

ramanav commented:
Removed some text content and properties which were used while debugging..

mdwragg commented:
Hi! So, the code all looks sane (nice job!), and the feature seems like it would be useful... Can you help us explain what the feature is capable of by extending index.html (the demo page) to explain to a developer how to use the feature?

Thanks!

ramanav commented:
we have a requirement to show the inline px-data-table again inside the container which we exposed. People can use it to display any card/component/chart/datatable inside   ##it.
![drilldownrowdetails](https://cloud.githubusercontent.com/assets/23207396/19980736/2d61fa3a-a225-11e6-8a49-71cd91064763.JPG)
![ourrequirement](https://cloud.githubusercontent.com/assets/23207396/19980737/2d65e99c-a225-11e6-82d7-f5b4dec06fa3.JPG)

mdwragg commented:
Cool, I understand now! Can you please extend the index.html to demonstrate to a developer how they would use this feature? 

When I ran up the test fixture, clicking on the rows didn't open up anything, should it?

Thanks.

ramanav commented:
Updated both index.html and demo.html to show how expanded property works.

mdwragg commented:
Hi, thanks for the additions to demonstrate your changes - can you show how you would implement the behaviour you specified in the design that you shared (above ^^). I can see how the toggle switches opens all rows that have 'expanded' content but not on a per-row basis as per your design.

Thanks!

ramanav commented:
Added row expand event listener to demonstrator how the row expand works. We were not able to get the datatable inside web component ready so added onclick for the toggle switch but in real time we dont need this we can add the listener on web component ready.

mdwragg commented:
Ah, sadly the changes dont work in shadow DOM - try running in Chrome and force shadow DOM mode with:

http://localhost:8080/?dom=shadow

Thanks.

mdwragg commented:
Closing as the customer no longer needs the change

bms-spog commented:
The Plant Calendar team would like to make use of this feature. It seems like it's almost complete, is there an ETA on when it would be finished if it were to be reopened?

mdwragg commented:
Sadly the author of this PR didn't need it anymore. Happy for you to take up the charge though if it is something that you need!
